Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyRowWidget.cpp

```diff
@@ -1722,6 +1747,158 @@ namespace AzToolsFramework
 
         return m_parentRow->CanChildrenBeReordered();
     }
+
+        int PropertyRowWidget::GetIndexInParent() const
+    {
+        if (!GetParentRow())
+        {
+            return -1;
+        }
+
+        for (int index = 0; index < GetParentRow()->GetChildRowCount(); index++)
+        {
+            if (GetParentRow()->GetChildrenRows()[index] == this)
+            {
+                return index;
+            }
+        }
+
+        return -1;
+    }
+
+    bool PropertyRowWidget::CanMoveUp() const
+    {
+        if (!CanBeReordered())
+        {
+            return false;
+        }
+
+        PropertyRowWidget* firstChildOfParent = m_parentRow->GetChildRowByIndex(0);
+        if (this == firstChildOfParent)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    bool PropertyRowWidget::CanMoveDown() const
+    {
+        if (!CanBeReordered())
+        {
+            return false;
+        }
+
+        AZ::u32 numChildrenOfParent = m_parentRow->GetChildRowCount();
+        PropertyRowWidget* lastChild = m_parentRow->GetChildRowByIndex(numChildrenOfParent - 1);
+        if (this == lastChild)
+        {
+            return false;
+        }
+        return true;
+    }
+
+    int PropertyRowWidget::GetParentWidgetWidth()
+    {
+        QWidget* parent = parentWidget()->parentWidget()->parentWidget();
```

Comment:
Please add a comment here explaining why "parent = greatgrandparent" is correct. :)

Reply:
Added comments, renamed the method and changed the ancestor search method to make things clearer.